server/lib/chat/chat-service.ts

```diff
@@ -161,6 +164,21 @@ export default class ChatService {
     )
   }
 
+  private async banUserFromChannelIfNeeded(
+    channelId: SbChannelId,
+    targetId: SbUserId,
+    client: DbClient,
+  ): Promise<boolean> {
+    const count = await countBannedChannelIdentifiers({ channelId, targetId }, client)
+    if (count >= MIN_IDENTIFIER_MATCHES) {
+      await banUserFromChannel({ channelId, targetId, reason: 'ban evasion' }, client)
```

Comment:
This was fine for being exposed to admins, I would not do this for regular users. imo copy the previous ban reason (any of the previous ban reasons is fine, since there's no guarantee they'll all be the same, but perhaps picking the one with the most occurrences is best. This should be doable with a modification to the existing query), add a column to mark this as an automated enforcement (and mark automated enforcements in some way in the UI, so people aren't confused why they have users in the list they did not manually ban)